Framework/Kernel/src/TimeROI.cpp

```diff
@@ -210,36 +257,44 @@ std::vector<DateAndTime> TimeROI::getAllTimes(const TimeROI &other) {
   return times_all;
 }
 
-void TimeROI::replaceValues(const std::vector<DateAndTime> &times, const std::vector<bool> &values) {
-  if (times.size() != values.size()) {
-    std::stringstream msg;
-    msg << "Times and Values are different size: " << times.size() << " != " << values.size();
-    throw std::runtime_error(msg.str());
-  }
+void TimeROI::replaceROI(const TimeSeriesProperty<bool> *roi) {
```

Comment:
this function is quite permissive with the input `TimeSeriesProperty<bool> *roi`. I would add an optional argument `bool strict = false` to reject time series that do not conform to the `true, fase, true, false,..,true, false` sequence when `strict` is set to `true`.
Also, the docstring should explain the permissive `strict=false` mode.